### ISSUE TITLE
[Bug]:  Missing  db name for RenameCollection if db name of target collection is empty

### ISSUE THREAD

**Issue by jaime0815** [milvus-io/milvus#28908]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: master& 2.3
- Deployment mode(standalone or cluster):
- MQ type(rocksmq, pulsar or kafka):    
- SDK version(e.g. pymilvus v2.0.0rc2):
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

Target db name of the collection can't use db name within the current connection if the target db name of the collection is empty

**Comment by yanliang567** [comment]
discussed with @jaime0815, this is a refine fix for #27385

**Comment by yanliang567** [comment]
@NicoYuan1986 could you please help to verify the fix again? Also please check if our testcases need an update.
